### PR TITLE
Update DataValues/Serialization to use wikibase-codesniffer 1.2.0

### PR DESCRIPTION
**Pull request by rosalieper**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -23,13 +23,13 @@
 		"irc": "irc://irc.freenode.net/wikidata"
 	},
 	"require": {
-		"php": ">=5.5.9",
+		"php": ">=7.1",
```
Comment (itamargiv) [inline composer.json:26]: I think this should be 7.2

**Review by itamargiv — CHANGES_REQUESTED** [review]
Looks good so far. We should also probably change the `travis.yml` file if we want to make sure that CI passes onlt on versions we support

File: src/Deserializers/DataValueDeserializer.php [hunk 2]
```diff
@@ -14,14 +14,14 @@
...
+ * @license GPL-2.0-or-later
  * @author Jeroen De Dauw < jeroendedauw@gmail.com >
  * @author Thiemo Kreuz
  */
 class DataValueDeserializer implements DispatchableDeserializer {
 
-	const TYPE_KEY = 'type';
-	const VALUE_KEY = 'value';
+	private const TYPE_KEY = 'type';
+	private const VALUE_KEY = 'value';
```
Comment (JeroenDeDauw) [inline src/Deserializers/DataValueDeserializer.php:24]: Are you sure these are not used in some Wikibase code?

Comment (lucaswerkmeister) [reply src/Deserializers/DataValueDeserializer.php]: Good point, [code search](https://codesearch.wmcloud.org/search/?q=DataValueDeserializer%3A%3A(TYPE%7CVALUE)_KEY) finds a handful of uses in [`SnakDeserializer`](https://github.com/wmde/WikibaseDataModelSerialization/blob/master/src/Deserializers/SnakDeserializer.php#L130).

**Review by lucaswerkmeister — APPROVED** [review]

**Pull request merged**